# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/driver_utils.cpp
# Commit: 754436d8b3a2
# Full Hash: 754436d8b3a2023b90eb247b3a70335e74f2fdf6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-05-09 09:14:23
# Regressor Bug: 1633628
# File Overlap Count: 3
# Description:
#   Bug 1633628 - Vender: Don't use ClearView if we previously used dual source blending on Intel gen6. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D74137
# ==============================================================================

diff -r f8fd6f0a5102 -r 754436d8b3a2 gfx/angle/checkout/src/libANGLE/renderer/driver_utils.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/driver_utils.cpp	Fri May 08 18:39:28 2020 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/driver_utils.cpp	Fri May 08 19:28:53 2020 +0000
@@ -22,6 +22,9 @@
 // Referenced from https://cgit.freedesktop.org/vaapi/intel-driver/tree/src/i965_pciids.h
 namespace
 {
+// gen6
+const uint32_t SandyBridge[] = {0x0102, 0x0106, 0x010A, 0x0112, 0x0116, 0x0122, 0x0126};
+
 // gen7
 const uint32_t IvyBridge[] = {0x0152, 0x0156, 0x015A, 0x0162, 0x0166, 0x016A};
 
@@ -92,6 +95,11 @@
     return !(*this < version);
 }
 
+bool IsSandyBridge(uint32_t DeviceId)
+{
+    return std::find(std::begin(SandyBridge), std::end(SandyBridge), DeviceId) != std::end(SandyBridge);
+}
+
 bool IsIvyBridge(uint32_t DeviceId)
 {
     return std::find(std::begin(IvyBridge), std::end(IvyBridge), DeviceId) != std::end(IvyBridge);